addons/website_event/static/src/interactions/website_event_slot_details.js

```diff
@@ -0,0 +1,141 @@
+        },
+        "li.page-item": {
+            "t-on-click.prevent.stop": this._onChangePageClick,
+        },
+        ".o_wevent_slot_btn": {
+            "t-on-click.prevent.stop": this._onSlotSelected,
+        },
+        ".o_wevent_slot_btn_cancel, .btn-close": {
+            "t-on-click": this._onClose,
+        },
+    };
+
+    get selectedSlot() {
+        return this.form.getAttribute("data-selected-slot-id");
+    }
+
+    setup() {
+        this.currentSlotPage = 0;
+        this.datesLimitPerPage = 6;
+        this.form = this.el.querySelector("#slot_registration_form");
+        this.selectedSlotDatetime = this.el.querySelector(".o_wevent_selected_slot");
+        this.selectedSlotTitle = this.el.querySelector(".o_wevent_selected_slot_title");
+        // Init first slot page
+        if (this.el.querySelector(".pagination")) {
+            this._changePage(0);
+        }
+    }
+
+    /**
+     * Display the specified slot page. Index starting at 0.
+     * @param {Integer} numPage
+     */
+    _changePage(numPage) {
+        this.currentSlotPage = numPage;
+        const dates = this.el.querySelectorAll(".o_wevent_slot_date");
+        dates.forEach((date) => date.classList.add("d-none"));
+        const min = this.datesLimitPerPage * this.currentSlotPage;
+        const max = this.datesLimitPerPage * (this.currentSlotPage + 1);
+        Array.from(dates).slice(min, max).forEach(
+            date => date.classList.remove("d-none")
+        );
+        // Handle previous/next buttons display
+        const previousBtn = this.el.querySelector(".o_wevent_slot_previous button");
+        const nextBtn = this.el.querySelector(".o_wevent_slot_next button");
+        if (dates.length <= this.datesLimitPerPage) {
+            [previousBtn, nextBtn].forEach((button) => button.classList.add("d-none"));
+        } else {
+            [previousBtn, nextBtn].forEach((button) => {
+                button.classList.remove("disabled");
+            });
+            if (this.currentSlotPage === 0) {
+                previousBtn.classList.add("disabled");
+            } else if(max >= dates.length) {
+                nextBtn.classList.add("disabled");
+            }
+        }
+    }
+
+    /**
+     * Update the displayed slots page.
+     * @param {MouseEvent} ev
+     */
+    _onChangePageClick(ev) {
+        const numPage = ev.currentTarget.classList.contains("o_wevent_slot_next") ? this.currentSlotPage + 1 : this.currentSlotPage - 1;
+        const maxNumPage = Math.ceil(this.el.querySelectorAll(".o_wevent_slot_date").length / this.datesLimitPerPage) - 1;
+        if (numPage < 0 || numPage > maxNumPage) {
+            return;
+        }
+        this._changePage(numPage);
+    }
+
+    /**
+     * Reset slot selection
+     * @param {MouseEvent} ev
+     */
+    _onClose(ev) {
+        this.el.querySelectorAll(".o_wevent_slot_btn").forEach((btn) => btn.classList.replace("btn-primary", "btn-light"));
```

Comment:
Same here, probably possible to have a dynamic style on the item based on the fact that it's selected or not?